src/tree/items/ProjectReferencedPackageTreeItem.ts

```diff
@@ -1,14 +1,41 @@
+import { DependencyTree } from 'nuget-deps-tree/dist/src/DependencyTree/Tree';
+
 import { PackageReference } from "@core/Projects";
 import { TreeItem, TreeItemCollapsibleState, TreeItemContext, ContextValues } from "@tree";
 
 export class ProjectReferencedPackageTreeItem extends TreeItem {
-    constructor(context: TreeItemContext, pkgRef: PackageReference) {
-        super(context, pkgRef.name, TreeItemCollapsibleState.None, ContextValues.projectReferencedPackage, pkgRef.name);
+    constructor(context: TreeItemContext, private pkgRef: PackageReference, private projectNugetDeps: DependencyTree[], contextValue: string = ContextValues.projectReferencedPackage) {
+        super(
+            context,
+            pkgRef.name,
+            projectNugetDeps.length > 0 ? TreeItemCollapsibleState.Collapsed : TreeItemCollapsibleState.None,
+            contextValue,
+            pkgRef.name
+        );
         this.description = pkgRef.version;
         this.allowIconTheme = false;
         this.addContextValueSuffix();
     }
 
+    protected async createChildren(childContext: TreeItemContext): Promise<TreeItem[]> {
+        let result: TreeItem[] = [];
+        const newContext = this.context.copy(this.context.project, this);
+        if (newContext.parent) {
+            newContext.parent.id = newContext.parent.id + this.pkgRef.name;
```

Comment:
i think this is wrong as i'm mutating `parent`. would appreciate some advice here (basically we need to support nested packages with the same package name)

Reply:
with this `id` creation code, it should not be necessary:
```ts
id = this.parent.id + '-' + this.label + '[' + this.contextValue + ']';
```